src/test/java/org/example/integration/JobRoleControllerIT.java

```diff
@@ -1,69 +1,121 @@
-
 @ExtendWith(DropwizardExtensionsSupport.class)
 class JobRoleControllerIT {
 
-
-    static final DropwizardAppExtension<TestConfiguration> APP = new DropwizardAppExtension<>(
-            TestApplication.class, null,
-            new ResourceConfigurationSourceProvider()
-    );
+    static final DropwizardAppExtension<TestConfiguration> APP =
+            new DropwizardAppExtension<>(TestApplication.class, null, new ResourceConfigurationSourceProvider());
 
     @Test
-    public void authorization_givenAdminRole_whenJobRolesGET_shouldReturnStatus200(){
+    public void authorization_givenAdminRole_whenJobRolesGET_shouldReturnStatus200() {
         String adminToken = generateToken("email@example.com", 1);
         JwtUtils.validateToken(adminToken);
-        Integer result = APP.client().target("http://localhost:8080/api/job-roles").request()
-                .header("Authorization" , ("Bearer " + adminToken)).get().getStatus();
+
+        Integer result = APP.client()
+                .target("http://localhost:8080/api/job-roles")
+                .request()
+                .header("Authorization", ("Bearer " + adminToken))
+                .get()
+                .getStatus();
         assertEquals(200, result);
     }
 
     @Test
-    public void authorization_givenUserRole_whenJobRolesGET_shouldReturnStatus200(){
+    public void authorization_givenUserRole_whenJobRolesGET_shouldReturnStatus200() {
         String userToken = generateToken("email@example.com", 2);
         JwtUtils.validateToken(userToken);
-        Integer result = APP.client().target("http://localhost:8080/api/job-roles").request()
-                .header("Authorization" , ("Bearer " + userToken)).get().getStatus();
+
+        Integer result = APP.client()
+                .target("http://localhost:8080/api/job-roles")
+                .request()
+                .header("Authorization", ("Bearer " + userToken))
+                .get()
+                .getStatus();
         assertEquals(200, result);
     }
 
     @Test
-    public void authorization_givenNonAuthorizedUser_whenJobRolesGET_shouldReturnStatus401(){
-        Integer result = APP.client().target("http://localhost:8080/api/job-roles").request().get().getStatus();
+    public void authorization_givenNotAuthenticatedUser_whenJobRolesGET_shouldReturnStatus401() {
+        Integer result = APP.client()
+                .target("http://localhost:8080/api/job-roles")
+                .request()
+                .get()
+                .getStatus();
+        assertEquals(401, result);
+    }
+
+    @Test
+    public void applyForRole_givenNonAuthenticatedUser_shouldReturnStatus401() {
+
+        Integer result = APP.client()
+                .target("http://localhost:8080/api/job-roles/1/applications")
+                .request()
+                .post(Entity.entity(null, MediaType.MULTIPART_FORM_DATA))
+                .getStatus();
         assertEquals(401, result);
     }
 
+    @Test
+    @Disabled
```

Comment:
why disabled?

Reply:
We have no self-cleaning database for tests. This test will run green only once. It is left in case we would have a proper integration-tests database setup in the future.